poolscript/script.go

```diff
@@ -277,6 +437,30 @@ func IsExpirySpend(witness wire.TxWitness) bool {
 	return true
 }
 
+// IsTaprootExpirySpend determines whether the provided witness corresponds to
+// the expiration script path of a Taproot enabled (version 1) account.
+func IsTaprootExpirySpend(witness wire.TxWitness) bool {
```

Comment:
I added a quick test and a fuzz test for all the witness detection functions. I ran the fuzz test for a couple of million executions, looks good so far.

Reply:
I removed the fuzz test again, because that would mean we'd need to update to Go 1.18 in CI which requires a few other changes because of the linter. So I'll add that in a separate PR.